src/gui/CMakeLists.txt

```diff
@@ -1,13 +1,33 @@
 # Allow users to not build without the gui even if Qt is found
 option(BUILD_GUI "Build the GUI" ON)
 
-# If Qt is not installed there will not be cmake
-# support for the package so this needs to be "quiet".
-find_package(Qt5 QUIET COMPONENTS Core Widgets Gui Charts)
+# Allow users to build with the Charts widget
+option(CHARTS "Build GUI with Charts widget" OFF)
```

Comment:
Of course, but If I use `if (Charts_FOUND)` QtCharts is never found.

Reply:
It is possible to do:

`find_package(Qt5 QUIET COMPONENTS Core Widgets OPTIONAL_COMPONENTS Charts)`

so that `if(Qt5_FOUND)` will be true even if Qtcharts is not found. Then, check Qtchart using `if(TARGET Qt5::Charts)`.

It seems that Qt::Gui is not necessary.

Reply:
Sorry I had the syntax wrong but it looks like you've figured it out